evm/src/libraries/TrimmedAmount.sol

```diff
@@ -2,103 +2,121 @@
 
-    function eq(TrimmedAmount memory a, TrimmedAmount memory b) internal pure returns (bool) {
-        return a.amount == b.amount && a.decimals == b.decimals;
-    }
+    return TrimmedAmount.wrap(amount | dec);
+}
 
-    function gt(TrimmedAmount memory a, TrimmedAmount memory b) internal pure returns (bool) {
-        if (a.decimals != b.decimals) {
-            revert NumberOfDecimalsNotEqual(a.decimals, b.decimals);
-        }
+function getAmount(TrimmedAmount a) pure returns (uint64) {
+    // Extract the raw integer value from TrimmedAmount
+    uint72 rawValue = TrimmedAmount.unwrap(a);
 
-        return a.amount > b.amount;
-    }
+    // Right shift to keep only the higher 64 bits
+    uint64 result = uint64(rawValue >> 8);
+    return result;
+}
 
-    function lt(TrimmedAmount memory a, TrimmedAmount memory b) internal pure returns (bool) {
-        if (a.decimals != b.decimals) {
-            revert NumberOfDecimalsNotEqual(a.decimals, b.decimals);
-        }
+function getDecimals(TrimmedAmount a) pure returns (uint8) {
+    return uint8(TrimmedAmount.unwrap(a) & 0xFF);
+}
 
-        return a.amount < b.amount;
-    }
+function eq(TrimmedAmount a, TrimmedAmount b) pure returns (bool) {
+    return getAmount(a) == getAmount(b) && getDecimals(a) == getDecimals(b);
+}
 
-    function isNull(TrimmedAmount memory a) internal pure returns (bool) {
-        return (a.amount == 0 && a.decimals == 0);
+function checkDecimals(TrimmedAmount a, TrimmedAmount b) pure {
+    uint8 aDecimals = getDecimals(a);
+    uint8 bDecimals = getDecimals(b);
+    if (aDecimals != bDecimals) {
+        revert NumberOfDecimalsNotEqual(aDecimals, bDecimals);
     }
+}
 
-    function sub(
-        TrimmedAmount memory a,
-        TrimmedAmount memory b
-    ) internal pure returns (TrimmedAmount memory) {
-        if (a.decimals != b.decimals) {
-            revert NumberOfDecimalsNotEqual(a.decimals, b.decimals);
-        }
+function gt(TrimmedAmount a, TrimmedAmount b) pure returns (bool) {
+    checkDecimals(a, b);
+
+    return getAmount(a) > getAmount(b);
+}
+
+function lt(TrimmedAmount a, TrimmedAmount b) pure returns (bool) {
+    checkDecimals(a, b);
 
-        return TrimmedAmount(a.amount - b.amount, a.decimals);
+    return getAmount(a) < getAmount(b);
+}
+
+function sub(TrimmedAmount a, TrimmedAmount b) pure returns (TrimmedAmount) {
+    checkDecimals(a, b);
+
+    return packTrimmedAmount(getAmount(a) - getAmount(b), getDecimals(a));
+}
+
+function add(TrimmedAmount a, TrimmedAmount b) pure returns (TrimmedAmount) {
+    checkDecimals(a, b);
+
+    return packTrimmedAmount(getAmount(a) + getAmount(b), getDecimals(b));
+}
+
+function min(TrimmedAmount a, TrimmedAmount b) pure returns (TrimmedAmount) {
+    checkDecimals(a, b);
+
+    return getAmount(a) < getAmount(b) ? a : b;
+}
+
+library TrimmedAmountLib {
+    error AmountTooLarge(uint256 amount);
+
+    function unpackAmount(TrimmedAmount a) internal pure returns (uint64) {
```

Comment:
I would just drop this function (and `unpackDecimals`) -- don't see a need to not just use `getAmount` and `getDecimals` directly. would make the diff elsewhere smaller too